src/utils/sidebar.js

```diff
@@ -45,36 +58,16 @@ function findItem(data, targetPath) {
   }
 }
 
-function findChildWithSource(item) {
-  return item.source ? item : findChildWithSource(item.children[0])
-}
-
-function findPrevItemWithSource(data, item) {
-  if (item.source) {
-    return item
-  } else if (item.prev) {
-    const prevItem = findItem(data, item.prev)
+function findPrevItemWithSource(data, ref) {
+  if (ref && ref.source) {
+    return ref
+  } else if (ref && ref.prev) {
+    const prevItem = findItem(data, ref.prev)
 
     return findPrevItemWithSource(data, prevItem)
   }
 }
```

Comment:
Other than deleting `public/static/docs/user-guide/contributing/index.md` and updating `sidebar.json`, This is the only actual change that fixes the issue in question @shcheklein. FYI

Reply:
It's not even Next.js specific BTW, just some logic that node runs at load time. (One of the custom patterns I was mentioning.)

Reply:
not sure about `custom patterns` ... but yes, it's not Next-specific 👍

Reply:
I mean that we implement our own "static content optimization" by using a custom server.js that imports a "helper" (sidebar.js) that puts the nav sidebar structure in memory **at server load time**. I call this a custom pattern because Next has a feature called [Automatic Static Optimization](https://nextjs.org/docs/advanced-features/automatic-static-optimization) for similar purposes, but we can't use it because we need a custom server script for all the redirects and for the custom route to our docs engine (page). Related to #888